# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/AbstractRange.cpp
# Commit: b1f1e6ad93f8
# Full Hash: b1f1e6ad93f8d7634acfcd0d66467c6efc2ad80e
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2024-03-26 04:11:24
# Regressor Bug: 1867058
# File Overlap Count: 1
# Description:
#   Bug 1867058 - Part 5: Update nsRange to create and update mCrossBoundaryRange when necessary r=smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195305
# ==============================================================================

diff -r 02d89b880ac5 -r b1f1e6ad93f8 dom/base/AbstractRange.cpp
--- a/dom/base/AbstractRange.cpp	Mon Mar 25 13:40:57 2024 +0000
+++ b/dom/base/AbstractRange.cpp	Mon Mar 25 13:40:57 2024 +0000
@@ -341,6 +341,8 @@
       // which they have been collapsed to one end, and it also may have a pair
       // of start and end which are the original value.
       aRange->DoSetRange(aEndBoundary, aEndBoundary, newEndRoot);
+      aRange->AsDynamicRange()->CreateOrUpdateCrossShadowBoundaryRangeIfNeeded(
+          aStartBoundary, aEndBoundary);
     }
     return NS_OK;
   }